### PR TITLE
Add categoryNav endpoint

### PR DESCRIPTION
**Pull request by Justin-Holmes**
- Adds followed_category_ids to user model

https://www.pivotaltracker.com/story/show/154940047

### CODE REVIEW

**Comment by Justin-Holmes** [comment]
@alanpeabody Do you think I should test the response more here in terms of what properties we return?

File: apps/ello_v3/test/ello_v3/resolvers/user_categories_test.exs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      categoryNav {
+          id
+          name
+          slug
+          roshi_slug
+          order
+          header
+          description
+          cta_caption
+          cta_href
```
Comment (alanpeabody) [inline apps/ello_v3/test/ello_v3/resolvers/user_categories_test.exs]: I don't think we need a lot of these fields. I think we can remove roshi_slug to cta_href

Comment (alanpeabody) [reply apps/ello_v3/test/ello_v3/resolvers/user_categories_test.exs]: Actually, I would just remove it from the test, I have already removed from the type.

**Comment by alanpeabody** [comment]
@Justin-Holmes it may be worth testing the values as a form of regression testing, but LGTM besides removing those fields from the test query so our PRs can merge w/o issue.

File: apps/ello_v3/lib/ello_v3/resolvers/user_categories.ex [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  def call(_parent, %{current_user: nil} = args, _resolver) do
+    categories = args
+                 |> Map.merge(%{primary: true})
+                 |> Discovery.categories
+
+    {:ok, categories}
+  end
+  def call(_parent, %{current_user: current_user} = args, _resolver) do
+    categories = args
+                 |> Map.merge(%{ids: current_user.followed_category_ids})
```
Comment (alanpeabody) [inline apps/ello_v3/lib/ello_v3/resolvers/user_categories.ex]: If this is nil we don't throw an exception right?

Comment (alanpeabody) [reply apps/ello_v3/lib/ello_v3/resolvers/user_categories.ex]: Maybe the default takes care of it?

**Comment by alanpeabody** [comment]
👍

**Pull request merged**